### PR TITLE
Fix #474: allow vagrant localhost ssh in vm_devbox

### LINKED ISSUE #474: vm_devbox allow ssh connections from localhost
For https://github.com/radiasoft/sirepo/issues/6946

### PR DESCRIPTION
**Pull request by e-carlin**
Allow ssh connections from vagrant over localhost in vm_devbox. This allows services like sirepo.job_driver.sbatch to connect over localhost back in to the host via ssh to start a job_agent.

### CODE REVIEW

**Comment by e-carlin** [comment]
@robnagler you mentioned using unique passwords for vagrant instead of the default. Is that still something you think is necessary? I'll need to test that nothing breaks but I assume we can just set the password for the vagrant user when provisioning the vm.

**Comment by robnagler** [comment]
We should delete the password for the vagrant user and root. The user can then set the password when they need it. We should add a quick check on setup of Slurm that tells the user what to do (what env var to set, for example).

File: rsconf/component/vm_devbox.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from pykern.pkcollections import PKDict
 from pykern.pkdebug import pkdp
 from rsconf import component
+from rsconf import db
```
Comment (robnagler) [inline rsconf/component/vm_devbox.py:12]: We should switch to qualified imports `db` is too short.

File: rsconf/package_data/vm_devbox/start.sh.jinja [hunk 2]
```diff
@@ -79,6 +79,11 @@ PasswordAuthentication no
 PermitRootLogin no
 Protocol 2
 X11Forwarding yes
+
+# SSH acces from localhost (ex for sirepo.job_driver.sbatch)
```
Comment (robnagler) [inline rsconf/package_data/vm_devbox/start.sh.jinja:83]: Typo: access

**Review by robnagler — APPROVED** [review]

File: rsconf/package_data/vm_devbox/start.sh.jinja (hunk 2 above)
Comment (e-carlin) [reply rsconf/package_data/vm_devbox/start.sh.jinja]: https://github.com/radiasoft/rsconf/pull/475/commits/2eb3a2218adbc6b59588554bf0d3f8a996d68025

File: rsconf/component/vm_devbox.py (hunk 1 above)
Comment (e-carlin) [reply rsconf/component/vm_devbox.py]: https://github.com/radiasoft/rsconf/pull/475/commits/2eb3a2218adbc6b59588554bf0d3f8a996d68025

**Comment by e-carlin** [comment]
please review again with the addition of `passwd -d`

File: rsconf/package_data/vm_devbox/start.sh.jinja [hunk 3]
```diff
@@ -79,12 +79,20 @@ PasswordAuthentication no
...
+ListenAddress {{ this.local_ip }}:22
+Match User {{ this.run_u }} Address {{ this.local_ip }}
+    PasswordAuthentication yes
 EOF_INSTALL
 
 echo '{{ this.ssh_identity_pub_key }}' >> '/home/{{ this.run_u }}/.ssh/authorized_keys'
 
 install --mode=400 --owner=root --group=root /dev/stdin "{{ this.ssh_guest_host_key_f }}"<<EOF_INSTALL
 {{ this.ssh_host_key }}
+
```
Comment (robnagler) [inline rsconf/package_data/vm_devbox/start.sh.jinja:93]: Missing `EOF_INSTALL`

File: rsconf/package_data/vm_devbox/start.sh.jinja [hunk 4]
```diff
@@ -79,12 +79,20 @@ PasswordAuthentication no
...
+passwd -d {{ this.root_u }}
+passwd -d {{ this.run_u }}
 EOF_INSTALL
```
Comment (robnagler) [inline rsconf/package_data/vm_devbox/start.sh.jinja:96]: Remove

**Review by robnagler — CHANGES_REQUESTED** [review]

File: rsconf/package_data/vm_devbox/start.sh.jinja (hunk 4 above)
Comment (e-carlin) [reply rsconf/package_data/vm_devbox/start.sh.jinja]: https://github.com/radiasoft/rsconf/pull/475/commits/f6d73932d2eed3ee4acba92e60656c604e42060a

**Review by robnagler — APPROVED** [review]

**Pull request merged**